conventions/services/search.py

```diff
@@ -379,82 +382,169 @@ def _build_filters(self, queryset: QuerySet) -> QuerySet:
+                    )
+                )
+                .annotate(
+                    programme_ville_score=Case(
+                        When(
+                            programme_ville_similarity__gt=settings.TRIGRAM_SIMILARITY_THRESHOLD,
+                            then=Round(
+                                Coalesce(F("programme_ville_similarity"), 0.0),
+                                precision=2,
+                            ),
+                        ),
+                        default=Value(0.0),
+                    )
+                )
+                .annotate(
+                    programme_numero_score=Case(
+                        When(
+                            programme_numero_similarity__gt=settings.TRIGRAM_SIMILARITY_THRESHOLD,
+                            then=Round(
+                                Coalesce(F("programme_numero_similarity"), 0.0),
+                                precision=2,
+                            ),
+                        ),
+                        default=Value(0.0),
+                    )
+                )
+                .annotate(
+                    bailleur_nom_score=Case(
+                        When(
+                            bailleur_nom_similarity__gt=settings.TRIGRAM_SIMILARITY_THRESHOLD,
+                            then=Round(
+                                Coalesce(F("bailleur_nom_similarity"), 0.0),
+                                precision=2,
+                            ),
+                        ),
+                        default=Value(0.0),
+                    )
+                )
+                .annotate(
+                    bailleur_siret_score=Case(
+                        When(
+                            bailleur_siret_similarity__gt=settings.TRIGRAM_SIMILARITY_THRESHOLD,
+                            then=Round(
+                                Coalesce(F("bailleur_siret_similarity"), 0.0),
+                                precision=2,
+                            ),
+                        ),
+                        default=Value(0.0),
+                    )
+                )
+                .annotate(
+                    conv_numero_score=Case(
+                        When(
+                            conv_numero_similarity__gt=settings.TRIGRAM_SIMILARITY_THRESHOLD,
+                            then=Round(
+                                Coalesce(F("conv_numero_similarity"), 0.0),
+                                precision=2,
+                            ),
+                        ),
+                        default=Value(0.0),
+                    )
+                )
+                .annotate(
+                    score=F("programme_nom_score")
+                    + F("programme_ville_score")
+                    + F("programme_numero_score")
+                    + F("bailleur_nom_score")
+                    + F("bailleur_siret_score")
+                    + F("conv_numero_score")
+                )
```

Comment:
L'idée ici est de se créer un score unique, basé sur les résultats des similarités des trigrammes.
Si ils sont au dessus du seuil, on prend l'arrondi, sinon 0.
Le score est une addition de ces valeurs.

On pourrait également booster un critère, en multipliant l'arrondi obtenu.
Ex: le numéro de programme a plus de poids que le nom, on multplis par 2 ou on ajoute un boost ?
Qu'en pensez vous?

Reply:
Je me demande quelle différence on aurait entre prendre le max ou prendre la somme, mais la somme semble plus logique. Et en effet on peut envisager une pondération ici, potentiellement avec des variables d'env pour la changer on-the-fly.

Reply:
J'ai peur que le max ne permette pas d'ajouter des boost sur certains critères.
Exemple, je veux booster le nom de la ville, je fais +0.2 à `programme_ville_score` si il est supérieur au seuil. Est-ce que ça fonctionnerait avec un max?

Et je suis pas sûr que tous les scores sont normalisés et comparables.

> ... avec des variables d'env pour la changer on-the-fly ...

Oui c'est l'idée de cette méthode `_build_scoring` qui contiendrait la logique de scoring.
On peut imaginer par exemple avoir 2 algo différents et switcher de l'un à l'autre pour faire des tests. Mais on devrait deja attendre quelques retours et des tests à plus grande échelle.